### ISSUE TITLE
This is a very good component. Do you have any plans to migrate to Polymer 2.0-rc?

### ISSUE THREAD

**Issue by phani1kumar** [chuckh/polymer-quill#13]
Hi,

I tried to change the bower dependencies to 2.0-rc2 and tested, but the component is not working as expected. Can you please confirm if there are any plans to upgrade this to 2.0-rc?

Thanks and Best Regards,
Phani

**Comment by chuckh** [comment]
I plan to upgrade to Polymer 2.0, but will not have a chance to work on it until April.  Hopefully by then the 1.x to 2.0 conversion utility will be available .

**Comment by phani1kumar** [comment]
Thank you very much for such a quick response. I was able to make the html-render example work. But rest of the demos are not working. 

I am currently developing a product for which the polymer-quill-bubble-firebase and polymer-quill-html-render seems to be a perfect fit, I'll try to build a small miniature component for this functionality alone. Hope the conversion utility will be coming soon :)

**Comment by eob** [comment]
+1 Interested if there's a way to help the port to 2.0

**Comment by petecarapetyan** [comment]
+1 my js skills are too lame to offer much help but would be delighted to help test when it is time.

There are notes in the 2.0 docs about shady dom - which my implementation of polymer-quill uses, but not sure how to interpret.

**Comment by yorrd** [comment]
Yes, please make sure to tell us if you need help with something specific :) very interested in this too

**Comment by chuckh** [comment]
Sorry I have not been able to do this.  Help would be great,  @yorrd and @petecarapetyan you could fork _polymer-quill_ and try running `polymer lint --fix` and see what it does.  If it works create a pull request, if not report back on the problems.  See more below about `polymer lint --fix`. 

From @rictic 
> In the process of developing the upgrader we realized that it had a ton of overlap with the new linter (registry, configuration, etc). So the plan is to be able to run polymer lint to discover any issues your code has with your chosen version of Polymer (1.0, 2.0, 2.0-hybrid), and to be able to run polymer lint --fix to automatically fix as many of those issues as have unambiguous fixes.
> 
> You can try out the new linter by running polymer experimental-lint in polymer-cli@next, and in the next couple of weeks it will replace the old linter, and run as polymer lint.

see this discussion at: https://github.com/Polymer/polymer-upgrade/issues/4#issuecomment-285787035

**Comment by DannyMcwaves** [comment]
polymer 2 port for component available at https://github.com/DannyMcwaves/polymer-quill

**Comment by giona69** [comment]
@DannyMcwaves going to test it if you interested

**Comment by Hammond2789** [comment]
@giona69 It did not work at all for me. Same error "Uncaught SyntaxError: Unexpected identifier".

**Comment by giona69** [comment]
@Hammond2789 no it didn't work. What I did, it is to get https://github.com/quilljs/quill build it locally, and then replace the css and js in polymer-quill.html ... it works but you have to get the fork they are working on for the shadowRoot problem ... it works almost everywhere, except safari, see https://github.com/quilljs/quill/pull/1805

**Comment by chuckh** [comment]
I have created branch for pure Polymer 2.x at https://github.com/chuckh/polymer-quill/tree/polymer-2.x.  Use this to do Polymer 2.x work for polymer-quill.  This does NOT work yet.

@Hammond2789 could you try this branch with quill fork https://github.com/quilljs/quill/pull/1805.  See if it works for you.  I have enabled ShadyDOM so it should work for Safari too.

## Status
[] Use ShadowDOM Quill fork and test
[] Change to ShadowDOM Quill fork if it works
[] Get Bubble popup working
[] Get Snow toolbar working
[x] Demos displaying Snow toolbar
[x] Demos display content correctly
[x] Create Polymer 1.x branch for Polymer 1.x version
[x] Test with Quill 1.3.4 (DOES NOT WORK)
[x] Test with @chrismbeckett's fork (https://github.com/quilljs/quill/compare/develop...arsnebula:develop) - DOES NOT WORK
[x] Update Readme
[x] Change dependencies to Polymer 2.x and Polymer 2.x elements
[x] Update Demo files for Polymer 2.x syntax
[x] Add `<custom-style></custom-style>` where needed
[x] Added `<dom-bind></dom-bind>` where needed
[x] Change `@apply(--)` to `@apply --`
[x] Remove `:host ::content` prefix from styles files
[x] Add `<link rel="import" href="../polymer/lib/utils/async.html">` import where needed
[x] Change to  <script src="../../webcomponentsjs/webcomponents-lite.js" shadydom></script> for ShadyDOM for Polymer 2.x
[x] Update sample.txt content to load better
[x] Change HMTL Render to work with `inner-h-t-m-l` instead of `<marked-element>`
[x] Remove 

see all changes at https://github.com/chuckh/polymer-quill/compare/polymer-2.x?expand=1

**Comment by petecarapetyan** [comment]
Just now checking in with this thread. From reading the previous messages still not sure if I should try out one of the Polymer2 versions yet? If anything is ready to test out in next couple weeks I should be available, if it would be helpful. I looked at jumping in at the code level but my javascript skills are currently way sub-par for the level of work required. :(
